Sources/Sandbox.Game/Game/Entities/MyEntity.cs

```diff
@@ -100,6 +100,13 @@ public MyGameLogicComponent GameLogic
             set { Components.Add<MyGameLogicComponent>(value); }
         }
 
+		private HashSet<MyGameLogicComponent> m_gameLogicSet = null;
```

Comment:
I felt there was no need to needlessly enumerate when gamelogic had nothing attached to it.